collection_bucket.go

```diff
@@ -163,12 +163,12 @@ func (wh *CollectionBucket) DefaultDataStore() sgbucket.DataStore {
 	return wh.getOrCreateCollection(scopeAndCollection{defaultScopeName, defaultCollectionName})
 }
 
-func (wh *CollectionBucket) NamedDataStore(name sgbucket.DataStoreName) sgbucket.DataStore {
+func (wh *CollectionBucket) NamedDataStore(name sgbucket.DataStoreName) (sgbucket.DataStore, error) {
 	sc, err := newValidScopeAndCollection(name.ScopeName(), name.CollectionName())
 	if err != nil {
-		return nil
+		return nil, fmt.Errorf("attempting to create/update database with a scope/collection that is &s", err)
```

Comment:
should this be `%w` ?

```suggestion
		return nil, fmt.Errorf("attempting to create/update database with a scope/collection that is %w", err)
```